### PR TITLE
mertics: derive identity from TLS handshake

### PR DESCRIPTION
**Pull request by howardjohn**
Today we do an IP lookup. I think this should only fail when we are going through middle proxies or have `!inpod && !src_ip_spoofing` which is ~never.

For the middle proxy case we really should consider getting the other attributes from them as well (baggage) vs lookup the IP.. but at least TLS identity seems critical to get right + has a clear right answer on where to get the attribute from

### CODE REVIEW

File: src/proxy/metrics.rs [hunk 1]
```diff
@@ -153,16 +153,24 @@ impl CommonTrafficLabels {
         self
     }
 
+    fn with_derived_source_identity(mut self, w: Option<&DerivedWorkload>) -> Self {
+        let Some(w) = w else { return self };
+        // This is the identity from the TLS handshake; this is the most trustworthy source so use it
+        self.source_principal = w.identity.clone().into();
```
Comment (ilrudie) [inline src/proxy/metrics.rs:159]: should this handle if w is Some but w.identity is None? Seems like we may not want to overwrite with None.

Comment (howardjohn) [reply src/proxy/metrics.rs:159]: We actually might want to -- if the request is not over TLS I don't think we want to report the principal?

Let me check how envoy does it

Comment (ilrudie) [reply src/proxy/metrics.rs:159]: That would make sense. I'll add a hold so you can address or not but LGTM otherwise

**Review by ilrudie — APPROVED** [review]
LGTM aside from the question of proper source_principal behavior. feel free to drop the hold

File: src/proxy/metrics.rs (hunk 1 above)
Comment (howardjohn) [reply src/proxy/metrics.rs:160]: Envoy does NOT report the principal in metrics when doing plaintext

**Pull request merged**